model/version_activation.go

```diff
@@ -32,7 +39,83 @@ func DoProjectActivation(ctx context.Context, id string, ts time.Time) (bool, er
 	}
 
 	return activated, nil
+}
+
+func activateEveryRecentMainlineCommitForProject(ctx context.Context, projectRef *ProjectRef, ts time.Time) (bool, error) {
+	lastActivatedVersion, err := VersionFindOne(ctx, VersionByMostRecentActivated(projectRef.Id, ts))
+	if err != nil {
+		return false, errors.Wrap(err, "finding most recently activated version")
+	}
+
+	var activateVersions []Version
+	if lastActivatedVersion == nil {
+		// No previously activated versions - this might be a new project or first activation
+		// Activate ALL unactivated non-ignored versions to ensure complete coverage
```

Comment:
"ALL" isn't quite true, right, since we're still adhering to the limit? I might reword this